flow360/plugins/report/report_items.py

```diff
@@ -1220,21 +1266,34 @@ def get_doc_item(self, context: ReportContext, settings: Settings = None) -> Non
         self._handle_grid_input(context.cases)
         self._handle_title(context.doc, context.section_func)
         cases = self._filter_input_cases(context.cases, context.case_by_case)
+        self._check_2d_caption_validity(cases)
 
         file_names, x_lab, y_lab = self._get_figures(cases, context)
 
-        caption = NoEscape(f'{bold(y_lab)} against {bold(x_lab)} for {bold("all cases")}.')
-
         if self.items_in_row is not None:
+            if self.caption != "":
+                caption = NoEscape(self._handle_2d_caption())
+            else:
+                caption = NoEscape(f'{bold(y_lab)} against {bold(x_lab)} for {bold("all cases")}.')
```

Comment:
I would move all this logic to `_handle_2d_caption()`. If self.caption is not provided, then fallback to default which is constructed inside `_handle_2d_caption()`.